### PR TITLE
Add tests for Event

### PR DESCRIPTION
**Pull request by yassere**
I tried to write these tests in a way that demonstrates how all the Event methods work.

I'm not sure about how I used `setTimeout` to test `wait`. Is that ok or is it fragile?

### CODE REVIEW

File: packages/@romejs/events/Event.test.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+		t.is(third, "bar returns");
+		t.is(fourth, "foo returns");
+
+		t.looksLike(fooCalls, ["hello", "rome", "hi"]);
+		t.looksLike(barCalls, ["rome", "test", "hi"]);
+	},
+);
+
+test(
+	"Event 4",
```
Comment (sebmck) [inline packages/@romejs/events/Event.test.ts:112]:
```suggestion
	"Event#callSync with promise subscription",
```

File: packages/@romejs/events/Event.test.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+		// send has no return value
+		t.is(sendRet, undefined);
+
+		t.looksLike(fooCalls, ["hello", "rome"]);
+	},
+);
+
+test(
+	"Event 3",
```
Comment (sebmck) [inline packages/@romejs/events/Event.test.ts:70]:
```suggestion
	"Event subscription order",
```

File: packages/@romejs/events/Event.test.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+		t.throws(() => {
+			event.callSync("error");
+		});
+
+		t.looksLike(fooCalls, ["hello", "rome"]);
+	},
+);
+
+test(
+	"Event 2",
```
Comment (sebmck) [inline packages/@romejs/events/Event.test.ts:46]:
```suggestion
	"Event send void",
```

File: packages/@romejs/events/Event.test.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
+import Event from "./Event";
+import {test} from "rome";
+
+type Callback<Param, Ret> = (param: Param) => Ret | Promise<Ret>;
+
+test(
+	"Event 1",
```
Comment (sebmck) [inline packages/@romejs/events/Event.test.ts:7]:
```suggestion
	"Event basic",
```

**Comment by sebmck** [comment]
Thank you! My suggested fixed caused some lint failures but I'll fix those in `master`.

File: packages/@romejs/events/Event.test.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+		const event = new Event<string, string>({name: "testEvent"});
+		let foo: string;
+
+		setTimeout(
+			() => {
+				foo = event.callSync("wait for this");
+				t.is(foo, "wait arg");
+			},
+			100,
+		);
```
[the remark this answers is not in the document]
Comment (yassere) [reply packages/@romejs/events/Event.test.ts]: I'm getting a `Test worker was unresponsive for 10 seconds` error with this new version. I think that by awaiting the new Promise, `event.callSync()` gets called before the `event.wait()` subscription is made so it throws an error and the promise doesn't resolve.

In the old version, I don't really understand why the test would end if it's awaiting `event.wait("wait arg")`. When I change the `t.is()` assertions to inlineSnapshots, they both fill out correctly: `t.inlineSnapshot(foo, "wait arg");` and `t.inlineSnapshot(waitValue, "wait for this");`

**Pull request merged**